internal/controller/databasecluster_controller.go

```diff
@@ -863,7 +863,7 @@ func (r *DatabaseClusterReconciler) ReconcileWatchers(ctx context.Context) error
 	log := log.FromContext(ctx)
 	addWatcher := func(dbEngineType everestv1alpha1.EngineType, obj client.Object) error {
 		sources := []source.Source{
-			source.TypedKind(r.Cache, obj, &handler.EnqueueRequestForObject{}),
+			source.Kind(r.Cache, obj, &handler.EnqueueRequestForObject{}),
```

Comment:
Not related to this bug, but just to make it consistent with the other configurations ...